src/main/java/hudson/plugins/release/pipeline/ReleaseQueueListener.java

```diff
@@ -0,0 +1,23 @@
+package hudson.plugins.release.pipeline;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.model.Queue;
+import hudson.model.queue.QueueListener;
+
+/**
+ * copied from org.jenkinsci.plugins.workflow.support.steps.build.BuildQueueListener
+ */
+@Extension
+public class ReleaseQueueListener extends QueueListener {
```

Comment:
Nice2 have: add `@Restricted(NoExternalUse.class)` to classes, which are not supposed to be externally used. This extension point is rather temporary IMHO